src/Quartz/Impl/AdoJobStore/UpdateRowLockSemaphore.cs

```diff
@@ -129,7 +129,11 @@ public UpdateLockRowSemaphore(IDbProvider provider)
 
                 if (await cmd.ExecuteNonQueryAsync().ConfigureAwait(false) != 1)
                 {
-                    throw new DataException(
+                    // TODO (NetCore Port): I think keeping the exception type consistent between desktop and Core versions of Quartz
+                    //                      is more important than keeping the desktop one consistent between the previous version and this one.
+                    //                      It's not an obvious call, though, so may be worth discussing.
+                    // throw new DataException(
+                    throw new InvalidOperationException(    // Maybe SqlException instead?
```

Comment:
Ah, I just realized DbException is an abstract class. We're not meant to throw it directly. Perhaps throwing InvalidOperationException here is good enough.